lib/CL/devices/common.c

```diff
@@ -1166,8 +1165,10 @@ pocl_check_kernel_dlhandle_cache (_cl_command_node *command,
 
   char module_fn[POCL_MAX_PATHNAME_LENGTH];
   int err = pocl_check_kernel_disk_cache (module_fn, command, specialize);
-  if (err)
+  if (err) {
+    POCL_UNLOCK (pocl_dlhandle_lock);
     return NULL;
+  }
```

Comment:
Nit. Probably should be:
```suggestion
  if (err) 
    {
      POCL_UNLOCK (pocl_dlhandle_lock);
      return NULL;
    }
```